### ISSUE TITLE
Question about back navigation in simulator

### ISSUE THREAD

**Issue by omaralbeik** [nalexn/clean-architecture-swiftui#4]
Thank you for the project @nalexn!

Could you please explain the following check for simulator 🤔

https://github.com/nalexn/clean-architecture-swiftui/blob/96af019bd581576531d4f0cd29fc34bcc629ebc1/CountriesSwiftUI/UI/Screens/CountryDetails.swift#L30

It is causing the details page not to pop back if it is presented by tapping a parent details page

**Comment by nalexn** [comment]
Hey Omar, there is a [bug in SwiftUI](https://stackoverflow.com/questions/58404725/why-does-my-swiftui-app-crash-when-navigating-backwards-after-placing-a-navigat) that basically crashes the app on the simulator when you try to pop the screen back.

**Comment by omaralbeik** [comment]
I see, looks like changing the navigation link to:
```swift
NavigationLink(destination: self.detailsView(country: country)) {
    CountryCell(country: country)
}
```
is solving this particular issue, but then tapping on the last pushed country in the list after dismissing details is not showing details

**Comment by nalexn** [comment]
Ok, the bug has been fixed in iOS 13.5 and so I could finally remove that workaround. Closing this issue.
